moto/s3/models.py

```diff
@@ -115,15 +116,14 @@ def append_to_value(self, value):
         self.last_modified = datetime.datetime.utcnow()
         self._etag = None  # must recalculate etag
         if self._is_versioned:
-            self._version_id += 1
-        else:
-            self._is_versioned = 0
```

Comment:
Note that this was very likely to be a bug. I think the original author wanted to write `self._version_id = 0`.